mach_eval/analyzers/electromagnetic/bspm/jmag_2d.py

```diff
@@ -139,20 +139,39 @@ def l_coil(self):
     @property
     def R_coil(self):
         a_wire = (self.machine_variant.s_slot * self.machine_variant.Kcu) / (
-            2 * self.machine_variant.Z_q
+            self.machine_variant.no_of_layers * self.machine_variant.Z_q
         )
-        return (self.l_coil * self.machine_variant.Z_q * self.machine_variant.Q / 6) / (
+        return (self.l_coil * self.machine_variant.Z_q) / (
             self.machine_variant.coil_mat["copper_elec_conductivity"] * a_wire
         )
 
+    @property
+    def R_wdg(self):
+        return (self.R_coil * self.z_C)
+
+    @property
+    def m(self):
+        m = len(self.machine_variant.coil_groups)
+        return m
+
+    @property
+    def z_C(self):
+        if len(self.machine_variant.layer_phases) == 1:
+            z_C = self.machine_variant.Q / (2 * self.m)
+        elif len(self.machine_variant.layer_phases) == 2:
+            z_C = self.machine_variant.Q / (self.m)
+
+        return z_C
```

Comment:
Scary not to catch the unhandled `else` condition. Can you guarantee either 1 or 2 layers? Perhaps raise exception if not 1 or 2?

Also, can you just use `self.machine_variant.no_of_layers` instead of recomputing it here?